File: leetcode/KthSmallestElementinaBST_230.java
import java.io.*;
import lib.TreeNode;
import lib.ConstructTree;

class KthSmallestElement {

  static class Solution {
    public int kthSmallest(TreeNode root, int k) { 
      if (root.left != null) {
        kthSmallest(root.left, k);
      }
      System.out.println(root.val);

      if (root.right != null) {
        kthSmallest(root.right, k);
      }
      return k;
     }
  }

  public static void main(String[] args) throws IOException {
    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    String line;
    while ((line = in.readLine()) != null) {
      TreeNode root = ConstructTree.stringToTreeNode(line);
      line = in.readLine();
      int k = Integer.parseInt(line);
      
      int ret = new Solution().kthSmallest(root, k);
      String out = String.valueOf(ret);
      System.out.println(out);
    }
  }
}

File: leetcode/KthSmallestElementinaBST_230.cpp
#include "../lib/createTree.hpp"
using namespace std;

class Solution {
  int value;
  int count = 0;

public:
  int kthSmallest(TreeNode *root, int k) {
    TreeNode *curr = root;
    while(curr != nullptr) {
      if (curr->left) {
        
      }
    }
  }
};

int main() {
    string line;
    while (getline(cin, line)) {
        TreeNode* root = stringToTreeNode(line);
        getline(cin, line);
        int k = stringToInteger(line);
        
        int ret = Solution().kthSmallest(root, k);

        string out = to_string(ret);
        cout << out << endl;
    }
    return 0;
}